### PR TITLE
[VA-17071] Update Facility Locator with components

### PR DESCRIPTION
**Pull request by maxx1128**
## Summary

This is a series of component updates to what's being used in the facility locator. Some are being updated from v1 to v3, some v1 components are being brought in, and some v3 components are being brought in. All the specifics are in [ticket #17071.](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/17071)

* I've updated the existing components from v1 to v3.
* I've moved what elements over to components that I could, but not all were possible.
  * The input field "City, State, or Postal Code" won't work since it needs the "use my location" action link and the button absolutely positioned to clear the input.
  * The modal does not use an alert and cannot; however, it has alert styling built into it, so it is not needed.
  * Several anchor tags had more than plaintext or crashed some of the Cypress tests, so they couldn't use the `<va-link>` component.

## Related issue(s)

- _Link to ticket created in va.gov-cms repo_
[department-of-veterans-affairs/va.gov-cms#17071](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/17071)

## Testing done

Visual, see below screenshots

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |   Screen Shot 2024-03-13 at 3 23 17 PM  |  Screen Shot 2024-03-13 at 3 22 56 PM  |
| Desktop |  Screen Shot 2024-03-13 at 3 22 17 PM  |  Screen Shot 2024-03-13 at 3 22 08 PM  |

Note: I believe the mental health number not appearing for Orange VA Clinic in "after" is due to the main and mental health phone numbers being the same, and a change not yet pushed to prod filters out if they're the same.

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Comment by maxx1128** [comment]
The final tests blocking this ticket are `src/applications/facility-locator/tests/e2e/accessibility.cypress.spec.js`, but I haven't been able to find a workaround since the `cypress-plugin-tabs` currently doesn't let you tab into elements in the shadow DOM. Pausing this for now.

**Comment by jilladams** [comment]
@maxx1128 what was outcome of checking with @laflannery re: the shadow DOM and whether this PR should go ahead as is or pivot?

File: src/applications/facility-locator/components/search-results-items/Covid19Result.jsx [hunk 1]
```diff
@@ -81,14 +81,18 @@ const Covid19Result = ({
...
           </div>
         )}
         {infoURL && (
           <span className="vads-u-margin-top--2 vads-u-display--block">
-            <a href={infoURL} target="_blank" rel="noreferrer">
-              COVID-19 info at this location
-            </a>
+            <va-link
+              href={infoURL}
+              target="_blank"
```
Comment (eselkin) [inline src/applications/facility-locator/components/search-results-items/Covid19Result.jsx:92]: I'm not sure it respects the target and rel props.

**Review by eselkin — DISMISSED** [review]
LGTM

**Comment by laflannery** [comment]
@maxx1128 I am using this to test: http://3bd545180a26dd67af1e3d182b997884.review.vetsgov-internal/find-locations

Updates needed:
1. The breadcrumb component is present but empty:
Screenshot 2024-03-28 at 2 56 37 PM

2. When you select Community provider's - the facility type field is no longer aligned with the Service type field
Screenshot 2024-03-28 at 3 55 53 PM

3. The service type field is also not acting conditionally correctly as it should be. It should not be active for Vet Centers, Cemeteries or Community Pharmacies but it currently is - it just happens to be empty
Screenshot 2024-03-28 at 3 58 56 PM
5. Can you please unbold the input text for the City state field? Because this field is not a component it doesn't match the v3s
Screenshot 2024-03-28 at 3 49 59 PM

6. The Alerts used for the operating statutes should be using the "slim" version, meaning we need to add the `slim` prop to these.
Screenshot 2024-03-28 at 3 27 50 PM

7. Can you update the "Try using our other tools to search" link to use `<va-link>` component?
Screenshot 2024-03-28 at 4 25 24 PM

8. Can you update the search button to be a v3 `<va-button>` component?
Screenshot 2024-03-28 at 4 26 42 PM

9. Also, and I hate to do this - but I completely did not realize there was an alert being used when the Facility locator doesn't work. I saw it on the ngrok today when we were trying to test. This is a v1 component and should be updated to v3. Are we able to do this? I understand if not and if we should open a new ticket for this.
Screenshot 2024-03-28 at 2 25 01 PM

**Review by jilladams — DISMISSED** [review]
Adding a review to capture that Laura's review feedback is merge-blocking: https://github.com/department-of-veterans-affairs/vets-website/pull/28433#issuecomment-2026064419

**Comment by maxx1128** [comment]
@laflannery I made updates based on your feedback except for the below points:

8. The submit button didn't work as a `<va-button>` component since it broke the related functionality.
9. This specific alert isn't even using a v1 component and is some entirely custom markup. Moving that to a v3 component from that would be a separate ticket.

**Comment by jilladams** [comment]
@maxx1128 in this case, the current ticket for this PR includes componentizing custom code that isn't yet in a component. I left it that way, since we were already in PR. Can you speak to effort to adopt that component? We can def break out into a separate ticket since it's not on a deadline now, but if it's small to do and we can kill it here, let me know how that would affect your sprint?

**Comment by maxx1128** [comment]
> @maxx1128 in this case, the current ticket for this PR includes componentizing custom code that isn't yet in a component. I left it that way, since we were already in PR. Can you speak to effort to adopt that component? We can def break out into a separate ticket since it's not on a deadline now, but if it's small to do and we can kill it here, let me know how that would affect your sprint?

From what I saw, it didn't look like it would be easy to copy and paste it into the new component. I wasn't able to replicate that view locally, however. I'll look at it again and see if it could be knocked out relatively quickly.

**Comment by laflannery** [comment]
@maxx1128 3 more comments. 2 I think I missed during my first review which I apologize for and 1 is new:
1. The service type field should not be required. I confirmed on the current Prod Facility Locator
2.  The service type field doesn't have a border anymore on first load, Vet Centers, Cemeteries or Community Pharmacies:
Screenshot 2024-04-02 at 9 25 40 AM
3. When the Facility type field has an error, it's getting duplicate styles because the component has styles and the custom styles have also been retained:
Screenshot 2024-04-02 at 9 19 50 AM
   - We should remove the custom error styles and only use the component styles:
      - Remove class `usa-input-error` from the wrapping div
      - Remove the `va-select` element from the CSS `padding:0` declaration
      - Remove the `bor-rad` class from the `<va-select>` this will make the red line lok nicer when it appears.
Screenshot 2024-04-02 at 9 27 41 AM
Screenshot 2024-04-02 at 9 29 03 AM

**Comment by maxx1128** [comment]
@laflannery I addressed the second round of feedback as well. For point 2, that state without a border (`inert`) is the closest thing the v3 component has to a disabled state - I got around it by lowering the opacity and removing pointer events, which keeps users from clicking on it but doesn't remove keyboard events. If that doesn't work, we may need to stick with `inert` for now and make a ticket for the design system team to include the border styling in that state.

Screen Shot 2024-04-02 at 6 03 04 PM

I also made the changes you listed for point 3 and took a screenshot so you can more quickly check the changes.

Screen Shot 2024-04-02 at 6 08 00 PM

**Review by eselkin — CHANGES_REQUESTED** [review]
minor tweak, otherwise looks great. Breadcrumbs work. Search works.

**Comment by laflannery** [comment]
@maxx1128 I also tried many things to get the component to work for our use case but unfortunately it just doesn't. The best thing we can do right now is to go back to the custom implementation we had because that is functional, accessible and visually correct. Which I know isn't great because you worked hard but it's the best option for what we are working with. That being said, can you:
1. Put the custom implementation of the Service Type field back with 2 small adjustments:
   - Remove `class="bor-rad" style="font-weight: bold;"` form the `<select>` element because this will align the styles with the other fields
Screenshot 2024-04-04 at 12 48 54 PM
2. The error state for the Facility Type field is still not quite correct:
Screenshot 2024-04-04 at 12 51 01 PM
   - You need to remove the VADS padding that has been added to the component and then this should fix it:
Screenshot 2024-04-04 at 12 51 57 PM
Screenshot 2024-04-04 at 12 52 26 PM

**Review by eselkin — APPROVED** [review]
LGTM

**Review by laflannery — APPROVED** [review]
LGTM!

**Pull request merged**